reps/2023-03-20-air-storage-path.md

```diff
@@ -0,0 +1,155 @@
+## Summary
+
+We'd like to change how users of Ray AIR (specifically: Train + Tune) configure permanent storage locations.
+
+Current way:
+- Users can specify `RunConfig.local_dir` or `tune.run(local_dir)` for storage on the local node
+- Users can specify `SyncConfig.upload_dir` for storage on remote nodes
+
+Future way:
+- We have a unified setting `RunConfig.storage_path` that can be set either to a local dir or a remote path.
+- Setting the storage path to a cloud or NFS URI (e.g., `s3://`, or `file://` that points to a NFS mount). In these cases, data will be first written to a local cache dir on the worker, and then synced to a subdirectory in the storage path designated by `<experiment_name>/<trial_name>/`.
```

Comment:
not sure if users know how to differentiate and tell local disk from nfs.
we can't detect this ourselves?

Reply:
Currently the driver triggers a pull-based sync from the worker nodes to the head node.

Instead, we can move this to the trainable on the worker nodes in the same way that we do for cloud-based syncing today.

Then the driver can drop a marker file in the trial directory. The trainable can see if this marker exists - if so, no syncing is necessary as it's a shared directory (either the trial is running on the head node, or it's NFS). Syncing only has to be done if the file is not found.

I can add this to the REP.

Reply:
Yeah, actually RAY_STORAGE guarantees the configured storage is available across the cluster already with this trick. So if you are able to obtain a filesystem from Ray storage, you can assume it's a NFS / S3 backed, or on a single node.

Reply:
nice